src/services/projects.js

```diff
@@ -470,131 +527,129 @@ module.exports = class ProjectsHelper {
 				['id', 'value', 'has_entities', 'validations']
 			)
 
-			// Using forEach instead of for loop for entity type validations
-
-			entityTypes.forEach((entityType) => {
-				const fieldData = projectData[entityType.value]
+			//fetch task entityType validations
+			let taskEntityTypes = await entityModelMappingQuery.findEntityTypesAndEntities(
+				{
+					model: common.TASKS,
+					status: common.STATUS_ACTIVE,
+				},
+				userDetails.organization_id,
+				['id', 'value', 'validations', 'has_entities']
+			)
 
-				if (entityType.validations.required) {
-					let required = utils.checkRequired(entityType, fieldData)
-					if (!required) {
-						throw {
-							message: `${entityType.value} not added`,
-							error: utils.errorObject(common.BODY, entityType.value),
-						}
-					}
-				}
+			const taskEntityTypesMapping = taskEntityTypes.reduce((acc, item) => {
+				acc[item.value] = item
+				return acc
+			}, {})
 
-				if (entityType.has_entities) {
-					let checkEntities = utils.checkEntities(entityType, fieldData)
-					if (!checkEntities.status) {
-						throw {
-							message: checkEntities.message,
-							error: utils.errorObject(common.BODY, entityType.value),
-						}
-					}
-				}
-
-				if (entityType.validations.regex) {
-					let checkRegex = utils.checkRegexPattern(entityType, fieldData)
-					if (checkRegex) {
-						throw {
-							message: `Special characters not allowed in ${entityType.value}`,
-							error: utils.errorObject(common.BODY, entityType.value),
-						}
+			//validate project data
+			for (let entityType of entityTypes) {
+				let validationResult = await this.validateEntityData(
+					projectData,
```

Comment:
@priyanka-TL just pass the required entityType here, also can we do this in parallel?